src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/GetManifestAsAssetTests.cs

```diff
@@ -0,0 +1,204 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using FluentAssertions;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.DotNet.Maestro.Tasks.Tests.Mocks;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.Maestro.Tasks.Tests
+{
+    [TestFixture]
+    public class GetManifestAsAssetTests
+    {
+        private PushMetadataToBuildAssetRegistry pushMetadata;
+        private const string locationString = "https://url.test/accountName/ProjectName/_apis/build/builds/buildNumberHere/artifacts";
+        private const string newManifestName = "NewManifest";
+
+        internal static readonly AssetData PackageAsset1 =
+            new AssetData(true)
+            {
+                Locations = ImmutableList.Create(
+                    new AssetLocationData(LocationType.Container)
+                    { Location = locationString }),
+                Name = "PackageAsset1Name",
+                Version = "1.2.3-beta.1235.6"
+            };
+
+        internal static readonly AssetData ExpectedPackageAsset1 =
+            new AssetData(true)
+            {
+                Locations = ImmutableList.Create(
+                    new AssetLocationData(LocationType.Container)
+                    { Location = locationString }),
+                Name = $"assets/manifests/thisIsARepo/1.2.3-beta.1235.6/{newManifestName}",
+                Version = "1.2.3-beta.1235.6"
+            };
+
+        [SetUp]
+        public void SetupGetManifestAsAssetTests()
+        {
+            pushMetadata = new PushMetadataToBuildAssetRegistry
+            {
+                getEnvProxy = new GetEnvMock()
+            };
+        }
+
+        [Test]
+        public void GivenExistentAssetDataWithLocationAndManifest()
+        {
+            AssetData parsedAssets = pushMetadata.GetManifestAsAsset(ImmutableList.Create(PackageAsset1), locationString, newManifestName);
+            parsedAssets.Should().BeEquivalentTo(ExpectedPackageAsset1);
+        }
+
+        // TODO: Why does this take multiple assets? It throws most of them out without looking at them.
```

Comment:
It uses the set of assets to find a non-shipping asset as a fallback for if AssetVersion is not set. You could alter the code to either:
- Take a single asset that can be used as fallback
- Take a version that can be used as a fallback
- Require that AssetVersion be set when this method is called, and ensure that it is set up prior to making the call. I think this is my preferred approach. Maybe add an "EnsureAssetVersion" method which ensures that:
  - If AssetVersion is not set on the task, then check the input assets for a non-shipping asset and sets it to that version.
  - If the version is still not empty, error.

Reply:
Good to know, thanks! I don't think it needs to be altered necessarily, especially given that it works & we're trying to wrap up, I just wasn't sure why it was the case.